open-metrics-report/src/main.rs

```diff
@@ -0,0 +1,201 @@
+    );
+
+    metrics::describe_gauge!(
+        METRIC_SAMPLE_COUNT,
+        metrics::Unit::Count,
+        "Number of samples in last message received"
+    );
+
+    task::spawn(poll_kafka_msg(consumer));
+
+    Ok(())
+}
+
+/// Poll kafka messages and update digitiser data.
+async fn poll_kafka_msg(consumer: StreamConsumer) {
+    loop {
+        match consumer.recv().await {
+            Err(e) => log::warn!("Kafka error: {}", e),
+            Ok(msg) => {
+                log::debug!(
+                    "key: '{:?}', topic: {}, partition: {}, offset: {}, timestamp: {:?}",
+                    msg.key(),
+                    msg.topic(),
+                    msg.partition(),
+                    msg.offset(),
+                    msg.timestamp()
+                );
+
+                if let Some(payload) = msg.payload() {
+                    if digitizer_analog_trace_message_buffer_has_identifier(payload) {
+                        match root_as_digitizer_analog_trace_message(payload) {
+                            Ok(data) => {
+                                let id = data.digitizer_id();
+                                let labels = [("dynamic_key", format!("{}", id))];
+
+                                /* Metrics */
+                                // Message recieved count
+                                counter!("digitiser_message_received_count", &labels).increment(1);
+
+                                // Last message frame number
+                                let frame_number = data.metadata().frame_number();
+                                gauge!("digitiser_last_message_frame_number", &labels)
+                                    .set(frame_number as f64);
+
+                                // Channel count
+                                let channel_count = match data.channels() {
+                                    Some(c) => c.len(),
+                                    None => 0,
+                                };
+                                gauge!("digitiser_channel_count", &labels)
+                                    .set(channel_count as f64);
+
+                                // Sample count
+                                let num_samples_in_first_channel = match data.channels() {
+                                    Some(c) => c.get(0).voltage().unwrap().len(),
+                                    None => 0,
+                                };
+                                gauge!("digitiser_sample_count", &labels)
+                                    .set(num_samples_in_first_channel as f64);
+
+                                // Last message timestamp
+                                let timestamp: Option<GpsTime> =
+                                    data.metadata().timestamp().copied().map(|t| t.into());
+                                gauge!("digitiser_last_message_timestamp", &labels)
+                                    .set(timestamp.unwrap().nanosecond() as f64);
```

Comment:
What does `nanosecond()` actually return in this case?

Reply:
Just realised that this will be the nanosecond part of the timestamp (omitting the date, hours and minutes). Should this be the number of seconds since the Unix Epoch instead?

Reply:
Unix timestamp would be the best (and pretty much only in OpenMetrics) option for this, it would need nanosecond precision however. Representing this as a `f64` with the unit as seconds should be good enough.